From: Otto Kekalainen <otto@debian.org>
Date: Tue, 29 Jun 2021 21:08:05 -0700
Subject: Make RocksDB build reproducible

The RocksDB binary included a string with the build timestamp:
> rocksdb_build_git_date:@2021-05-23-16:04:38@

As this changes from build to build, it makes the builds unreproducible.
Simply removing it solves the issue.

This temporary fix can be removed when a proper fix already done in upstream
lands in MariaDB when the RocksDB submodule is updated to a newer release.

Origin: https://github.com/facebook/rocksdb/commit/0a9a05ae12943b1529ef1eabbca5ce5a71c986bf
Bug: https://github.com/facebook/rocksdb/issues/7035
# Merged in RocksDB 6.19.3, but not updated into MariaDB yet
---
 storage/rocksdb/rocksdb/util/build_version.cc.in | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/storage/rocksdb/rocksdb/util/build_version.cc.in b/storage/rocksdb/rocksdb/util/build_version.cc.in
index 8569580..56a1235 100644
--- a/storage/rocksdb/rocksdb/util/build_version.cc.in
+++ b/storage/rocksdb/rocksdb/util/build_version.cc.in
@@ -1,5 +1,5 @@
 // Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 #include "build_version.h"
-const char* rocksdb_build_git_sha = "rocksdb_build_git_sha:@@GIT_SHA@@";
-const char* rocksdb_build_git_date = "rocksdb_build_git_date:@@GIT_DATE_TIME@@";
-const char* rocksdb_build_compile_date = __DATE__;
+const char* rocksdb_build_git_sha = "rocksdb_build_git_sha:REDACTED";
+const char* rocksdb_build_git_date = "rocksdb_build_git_date:REDACTED";
+const char* rocksdb_build_compile_date = "REDACTED";
